### PR TITLE
21_1 Grids editing.changes: index and pageIndex description

### PR DESCRIPTION
**Pull request by pavel-2020**
[Trello](https://trello.com/c/9ZvrA4oB/1717-grids-editingchanges-describe-index-and-pageindex-fields)

### CODE REVIEW

File: api-reference/_hidden/DataChange/index.md [hunk 1]
```diff
@@ -4,7 +4,12 @@ type: Number
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+[tags] ctp
+
+Zero-based index of a new row.
```
Comment (RomanTsukanov) [inline api-reference/_hidden/DataChange/index.md]:
```suggestion
Zero-based index of a new row. Applies only if the [type]({currentpath}/#type) is *"insert"*.
```

File: api-reference/_hidden/DataChange/index.md [hunk 2]
```diff
@@ -4,7 +4,12 @@ type: Number
...
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+
+#include common-ctp-note-wo-devextreme 
+
+When pageIndex is used, sets the index relative to the specified page. To add a row to the bottom, assign -1 to this field.
```
Comment (RomanTsukanov) [inline api-reference/_hidden/DataChange/index.md]:
```suggestion
If you use the [pageIndex]({currentpath}/#pageIndex), this field sets the row index on the specified page. To add a row to the end of the page or dataset, set **index** to -1.
```

File: api-reference/_hidden/DataChange/pageIndex.md [hunk 3]
```diff
@@ -4,7 +4,12 @@ type: Number
...
+
+Zero-based page index of a new row. Applies if [pager]({basewidgetpath}/Configuration/pager/) is used.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+
+#include common-ctp-note-wo-devextreme 
+
+When not specified, equals the current page index. To add a row to the last page, set this field to -1.
```
Comment (RomanTsukanov) [inline api-reference/_hidden/DataChange/pageIndex.md]:
```suggestion
When not specified, this field equals the [paging]({basewidgetpath}/Configuration/paging/).[pageIndex]({basewidgetpath}/Configuration/paging/#pageIndex) value. To add a row to the last page, set this field to -1.
```

File: api-reference/_hidden/DataChange/pageIndex.md [hunk 4]
```diff
@@ -4,7 +4,12 @@ type: Number
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+[tags] ctp
+
+Zero-based page index of a new row. Applies if [pager]({basewidgetpath}/Configuration/pager/) is used.
```
Comment (RomanTsukanov) [inline api-reference/_hidden/DataChange/pageIndex.md]:
```suggestion
Zero-based index of a page into which a new row should be inserted. Applies only if the [type]({currentpath}/#type) is *"insert"* and the [pager]({basewidgetpath}/Configuration/pager/) is used.
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: api-reference/_hidden/DataChange/index.md (hunk 1 above)
Comment (jtoming830) [reply api-reference/_hidden/DataChange/index.md]: > on the [current page]

@RomanTsukanov what do you mean by 'current page' in the case with virtual scrolling? Viewport or batch of loaded data rows or something else?

Comment (RomanTsukanov) [reply api-reference/_hidden/DataChange/index.md]: @jtoming830 You tell me :) I consider the page specified by pageIndex to be the current page.

**Review by RomanTsukanov — APPROVED** [review]

File: api-reference/_hidden/DataChange/index.md [hunk 5]
```diff
@@ -4,7 +4,12 @@ type: Number
...
+
+Zero-based index of a new row. Applies only if the [type]({currentpath}/#type) is *"insert"*.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+
+#include common-ctp-note-wo-devextreme 
+
+If you use the [pageIndex]({currentpath}/#pageIndex), this field sets the row index on the specified page. To add a row to the end of the page or dataset, set **index** to -1.
```
Comment (RomanTsukanov) [inline api-reference/_hidden/DataChange/index.md]:
```suggestion
If the [pageIndex]({currentpath}/#pageIndex) has a value, this field sets a row index on the specified page. In other cases, this field specifies the row index within the entire grid. Only data and group rows are indexed. To add a row to the end of the page or grid, set **index** to -1.
```

File: api-reference/_hidden/DataChange/pageIndex.md [hunk 6]
```diff
@@ -4,7 +4,12 @@ type: Number
...
+
+Zero-based index of a page into which a new row should be inserted. Applies only if the [type]({currentpath}/#type) is *"insert"* and the [pager]({basewidgetpath}/Configuration/pager/) is used.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+
+#include common-ctp-note-wo-devextreme 
+
+When not specified, this field equals the [paging]({basewidgetpath}/Configuration/paging/).[pageIndex]({basewidgetpath}/Configuration/paging/#pageIndex) value. To add a row to the last page, set this field to -1.
```
Comment (RomanTsukanov) [inline api-reference/_hidden/DataChange/pageIndex.md]:
```suggestion
This field has a value if the DataGrid uses the pager. This value equals [paging]({basewidgetpath}/Configuration/paging/).[pageIndex]({basewidgetpath}/Configuration/paging/#pageIndex), but you can override it with a custom value. To add a row to the last page, set this field to -1.
```

File: api-reference/_hidden/DataChange/pageIndex.md [hunk 7]
```diff
@@ -4,7 +4,12 @@ type: Number
...
+
+Zero-based index of a page into which a new row should be inserted. Applies only if the [type]({currentpath}/#type) is *"insert"* and the [pager]({basewidgetpath}/Configuration/pager/) is used.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+
+#include common-ctp-note-wo-devextreme 
+
+This field has a value if the DataGrid uses the pager. This value equals [paging]({basewidgetpath}/Configuration/paging/).[pageIndex]({basewidgetpath}/Configuration/paging/#pageIndex), but you can override it with a custom value. To add a row to the last page, set this field to -1.
```
Comment (RomanTsukanov) [inline api-reference/_hidden/DataChange/pageIndex.md]:
```suggestion
This field has a value if the {WidgetName} uses the pager. This value equals [paging]({basewidgetpath}/Configuration/paging/).[pageIndex]({basewidgetpath}/Configuration/paging/#pageIndex), but you can override it with a custom value. To add a row to the last page, set this field to -1.
```

**Review by RomanTsukanov — APPROVED** [review]

**Review by jtoming830 — APPROVED** [review]

**Pull request merged**